lib/BackgroundJob/Tasks/CheckRequirementsTask.php

```diff
@@ -105,8 +97,37 @@ public function execute(FaceRecognitionContext $context) {
 				"If you already set this to unlimited, it seems your system is not having enough RAM memory.";
 			$this->logInfo($error_message);
 			return false;
+		} else {
+			$this->logDebug(sprintf('Found %d MB available to PHP.', $memory / (1024 * 1024)));
+			// No point going above 4GB ("4GB should be enough for everyone")
+			if ($memory > 4 * 1024 * 1024 * 1024) {
+				$this->logDebug('Capping used memory to maximum of 4GB');
+				$memory = 4 * 1024 * 1024 * 1024;
+			}
 		}
 
+		// Apply admin setting limit.
+		$memorySetting = $this->config->getAppValue('facerecognition', 'memory-limits', '-1');
+		if ($memorySetting > 0) {
+			if ($memorySetting > $memory) {
+				$error_message =
+					"\n" .
+					"Seems that you configured" . intval($memorySetting / (1024 * 1024)) . " MB of memory, however, this exede"
+					"detected as maximum for PHP: " . intval($memory / (1024 * 1024)). "MB.\n" .
+					"You need to change your memory_limit in php.ini or 'Memory Limits' in settings.\n" .
+					"Check https://secure.php.net/manual/en/ini.core.php#ini.memory-limit for details.\n" .
+					"If you already set this to unlimited, it seems your system is not having enough RAM memory.";
```

Comment:
> If you already set this to unlimited

As per this code, this cannot happen (this is under `($memorySetting > 0)`, right?

Reply:
Oh..  You're right, did not think well it, just copy other debug. :sweat_smile:

Reply:
I removed this. See if it's okay.